### PR TITLE
Cypress end-to-end testing

### PR DESCRIPTION
**Pull request by dmtrKovalenko**

### CODE REVIEW

File: lib/src/MuiPickersUtilsProvider.tsx [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import * as PropTypes from 'prop-types';
 import * as React from 'react';
 
-const { Consumer, Provider } = React.createContext(null as any);
-export const MuiPickersContextConsumer = Consumer;
+export const MuiPickersContext = React.createContext(null as any);
+// TODO remove in v3.0
+export const MuiPickersContextConsumer = MuiPickersContext.Consumer;
```
[the remark this answers is not in the document]
Comment (dmtrKovalenko) [reply lib/src/MuiPickersUtilsProvider.tsx]: @TrySound I see, I will change it. The issue is that we are not providing default value in `React.createContext` but it comes from the user and required. But if `null` is passed that means that user doesn't wrapped the component in `MuiPickersUtilsProvider` so we must show an error then. But still context is not nullable type :( 
`React.Context<IUtils>(null as any)` <- possible workaround

[the remark this answers is not in the document]
Comment (TrySound) [reply lib/src/MuiPickersUtilsProvider.tsx]: I mean like this. If you throw an error after null check than return later value will be not nullable already. Does it work in TS?
https://github.com/tizmagik/react-head/blob/master/src/HeadTag.js#L30

[the remark this answers is not in the document]
Comment (TrySound) [reply lib/src/MuiPickersUtilsProvider.tsx]: Context should be nullable and hidden from user in the end.

[the remark this answers is not in the document]
Comment (dmtrKovalenko) [reply lib/src/MuiPickersUtilsProvider.tsx]: Yep, agreed it will work

[the remark this answers is not in the document]
Comment (dmtrKovalenko) [reply lib/src/MuiPickersUtilsProvider.tsx]: Fixed. I am now exporting context to make things like `useContext(MaterialUIPickersContext)` work. Then we will provide custom hook `useUtils()` for this purpose

**Pull request merged**